lib/srv/db/postgres/engine.go

```diff
@@ -409,6 +409,23 @@ func (e *Engine) auditUserPermissions(session *common.Session, entries []events.
 	e.Audit.OnPermissionsUpdate(e.Context, session, entries)
 }
 
+// auditResult process backend wire messages and emit result event on
+// appropriate messages.
+func (e *Engine) auditResult(session *common.Session, pgMsg pgproto3.BackendMessage) {
+	var res common.Result
+
+	switch m := pgMsg.(type) {
+	case *pgproto3.CommandComplete:
+		res.AffectedRecords = uint64(pgconn.CommandTag(m.CommandTag).RowsAffected())
```

Comment:
Is `RowsAffected()` automatically populated for "select" queries by Postgres as well?

Reply:
Yes, it is populated for all the ones that return the affected rows (select, update, insert, delete, merge, move, fetch, and copy). If a command doesn't return the affected rows, it will return 0.